security/src/main/java/com/fincity/security/service/ClientPasswordPolicyService.java

```diff
@@ -0,0 +1,226 @@
+package com.fincity.security.service;
+
+import static com.fincity.nocode.reactor.util.FlatMapUtil.flatMapMono;
+import static com.fincity.security.service.SecurityMessageResourceService.CLIENT_PASSWORD_POLICY_ERROR;
+
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import org.jooq.types.ULong;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.stereotype.Service;
+
+import com.fincity.saas.commons.jooq.service.AbstractJOOQUpdatableDataService;
+import com.fincity.security.dao.ClientPasswordPolicyDAO;
+import com.fincity.security.dto.ClientPasswordPolicy;
+import com.fincity.security.jooq.tables.records.SecurityClientPasswordPolicyRecord;
+
+import reactor.core.publisher.Mono;
+
+@Service
+public class ClientPasswordPolicyService extends
+        AbstractJOOQUpdatableDataService<SecurityClientPasswordPolicyRecord, ULong, ClientPasswordPolicy, ClientPasswordPolicyDAO> {
+
+	@Autowired
+	private SecurityMessageResourceService securityMessageResourceService;
+
+	private final Set<Character> specialCharacters = new HashSet<>(
+	        Arrays.asList('~', '`', '!', '@', '#', '$', '%', '^', '&', '*', '(', ')', '_', '-', '+', '=', '{', '}', '[',
+	                ']', '|', '\\', '/', ':', ';', '\"', '\'', '<', '>', ',', '.', '?'));
+
+	@PreAuthorize("hasAuthority('Authorities.Client_Password_Policy_UPDATE')")
+	@Override
+	protected Mono<ClientPasswordPolicy> updatableEntity(ClientPasswordPolicy entity) {
+		return super.update(entity);
+	}
+
+	@Override
+	protected Mono<Map<String, Object>> updatableFields(ULong key, Map<String, Object> fields) {
+		return Mono.just(fields);
+	}
+
+	public Mono<Boolean> checkPasswordPolicyExists(ULong clientId) {
+		return this.dao.checkClientPasswordPolicyExists(clientId);
+	}
+
+	public Mono<ClientPasswordPolicy> getPolicyByClientId(ULong clientId) {
+		return this.dao.getByClientId(clientId);
+	}
+
+	public Mono<Boolean> checkAllConditions(ULong clientId, String password) {
+
+		return flatMapMono(
+
+		        () -> this.dao.getByClientId(clientId),
+
+		        passwordPolicy -> passwordPolicy != null ? checkAlphanumericExists(passwordPolicy, password)
+		                : Mono.empty(),
+
+		        (passwordPolicy, isAlphaNumberic) -> passwordPolicy.isAtleastOneSpecialChar()
+		                ? checkInSpecialCharacters(password).map(val -> val && isAlphaNumberic)
+		                : Mono.just(isAlphaNumberic),
+
+		        // re check here
+		        (passwordPolicy, isAlphaNumberic, isSpecial) ->
+
+				!passwordPolicy.isSpacesAllowed() && password.contains(" ") ?
```

Comment:
The logic should be if the password contains a space, then check if it is allowed or not.

Reply:
updated need to test.